### PR TITLE
[HW_12][ts_hw2 'Generics, classes' task_01, task_02]

### PR DESCRIPTION
**Pull request by Maximus2806**

### CODE REVIEW

File: js_hw10/task_02.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// 1. Напишите функцию, реализующую методы массивов map. Функции принимают на вход массив и колбэк. Используйте дженерик типы. 
+//    Затипизировать надо саму функцию и коллбэк.
+//    Создать реализацию функции map, принимающую массив чисел 1-5, возвращающую новый массив, 
+//    где каждый каждый элемент - это элемент исходного массива умноженный на его индекс
+//    Пример:
+//    map([1,2,3,4,5], callback) => [0,2,6,12,20]
+
+type TCallback<I, R> = (item: I) => R;
```
Comment (anatoly-karpovich) [inline js_hw10/task_02.ts]:
```suggestion
type TCallback<I, R> = (item: I, index: number, arr: I[]) => R;
```

File: js_hw10/task_02.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+function mapArray<I,R>(arr: I[], callback: TCallback<I,R>):R[] {
+    const result: R[] = [];
+    arr.forEach((item) => result.push(callback(item)));
+    return result
+};
+
+const testArr = [1,2,3,4,5];
+console.log (mapArray(testArr, (item) => testArr.indexOf(item)*item));
```
Comment (anatoly-karpovich) [inline js_hw10/task_02.ts]:
```suggestion
console.log (mapArray(testArr, (item, i, arr) => arr.indexOf(item)*item));
```

**Review by anatoly-karpovich — APPROVED** [review]

File: js_hw10/task_02.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+//   ["4", 4],
+//   ["4", 5], // повторяющийся ключ, значит это значение должно быть в результирующем объекте
+// ]);
+
+// console.log(result); //{ '1': 1, '2': 2, '3': 3, '4': 5 }
+
+function generateObject<T>(arrOfPairs: [string, T][]):{[key:string]: T} {
+    const result: {[key:string]: T} = {};
+    for(const pair of arrOfPairs){
+        result[pair[0]] = pair[1];
```
Comment (VladimirLQA) [inline js_hw10/task_02.ts]: As option
```suggestion
    for(const [key, value] of arrOfPairs){
        result[key] = value;
```

**Review by VladimirLQA — APPROVED** [review]

File: js_hw10/task_02.ts (hunk 3 above)
Comment (Maximus2806) [reply js_hw10/task_02.ts]: I agree, looks better! Done.

File: js_hw10/task_02.ts (hunk 1 above)
Comment (Maximus2806) [reply js_hw10/task_02.ts]: Done

**Pull request merged**